ITL-3 Implement edit and add state for todo card

EmanuelRusuu opened the pull request.

EmanuelRusuu committed:
ITL-14 Subtask : Create an empty to-do item

- on click on the add button, an empty todo will appear

EmanuelRusuu committed:
ITL-15 Subtask 2: Display the new to-do item at the top of the list

- on click on the add button, the new added todo will appear first

EmanuelRusuu committed:
ITL-16 Subtask 3: Set proper placeholder text for the new to-do item

- proper placeholders were set on the new todo item

EmanuelRusuu committed:
ITL-17 Subtask 4: After tapping the task, Card will convert into edit mode

- on click on the todo, it converts into edit mode

EmanuelRusuu committed:
ITL-18 Subtask 5: Priority change todo

- only when the task is under editing mode, the user can extend the priority tab, and select/change the desired priority for the todo

EmanuelRusuu committed:
ITL-19 Subtask 6: Make the design match with figma

- improved the last subtask (ITL-18)
- design matches with figma

EmanuelRusuu committed:
ITL 20 Subtask 7: Pop-up confirmation

- improved last commit/subtask
- added text for mobile tasks
- added popup when deleting a task

roli-wg commented on todoapp/src/components/DeletePopUP.vue:
```diff
@@ -0,0 +1,206 @@
+<template>
+  <div class="popup-main-container">
+    <div class="popup-second-container">
+      <div class="x" @click="togglePopUP">
```
Please try to add more meaningful class names

roli-wg commented on todoapp/src/components/DeletePopUP.vue:
```diff
@@ -0,0 +1,206 @@
+          aria-hidden="true"
+        >
+          <path stroke-linecap="round" stroke-linejoin="round" d="M6 18L18 6M6 6l12 12"></path>
+        </svg>
```
Can you use an icon library like [Heroicons](https://heroicons.com/) for the icons or if you don't want please save the svg in separate file or create a vue component for it

roli-wg commented on todoapp/src/components/DeletePopUP.vue:
```diff
@@ -0,0 +1,206 @@
+          stroke-linecap="round"
+          stroke-linejoin="round"
+          d="M12 9v3.75m-9.303 3.376c-.866 1.5.217 3.374 1.948 3.374h14.71c1.73 0 2.813-1.874 1.948-3.374L13.949 3.378c-.866-1.5-3.032-1.5-3.898 0L2.697 16.126zM12 15.75h.007v.008H12v-.008z"
+        ></path>
```
Same here

roli-wg commented on todoapp/src/components/TodoList.vue:
```diff
@@ -2,44 +2,65 @@
 }
-
 onMounted(() => {
   todos.value = JSON.parse(localStorage.getItem('todos')!) || []
```
Please avoid as much as you can using "!" at the end of expressions this doesn't solve any error just say to TS compiler that you are sure that the value you try to access will exists. This is a good example why to not use this "trick" if in your localStorage doesn't exist the todos attribute you will receive an error because JSON cannot parse undefined
You can simple check if the `localStorage.getItem('todos')` exist and than parse

roli-wg commented on todoapp/src/components/TodoList.vue:
```diff
@@ -2,44 +2,65 @@
+  <div class="delete-popup">
+    <DeletePopUP
+      v-if="popUp"
+      @togglePopUP="togglePopUP"
```
Please use hyphenated format for all your props and events and respect the order of attributes props goes first and the emit ( move `:index="selectedTaskIndex"` before your events)

EmanuelRusuu committed:
Solved git changes request

- used more descriptive class names
- created components for svg`s
- removed the ! at getting the 'todos' from localStorage
- switched props and emit position

komaromiGyorgy commented on todoapp/.vite/deps/_metadata.json:
Please put the .vite folder into gitignore

komaromiGyorgy commented on todoapp/src/components/Icons/AddIcon.vue:
```diff
@@ -0,0 +1,29 @@
+<!-- eslint-disable vue/multi-word-component-names -->
```
You can add the rule: vue/multi-word-component-names into the eslintrc file, this way you do not need to specify it on top of files.

EmanuelRusuu committed:
Solved git changes request

- Added line to ignore single-word components in ESLint
- Updated .gitignore for .vite folder

roli-wg commented on todoapp/src/components/DeletePopUP.vue:
```diff
@@ -0,0 +1,168 @@
+<script setup lang="ts">
+import { defineProps, defineEmits } from 'vue'
```
You don't need to import defineProps and defineEmits are built in macros which can be accessed without importing them
You may also receive a warning in the terminal when you start the app saying the same, please make sure you don't have any errors or warnings in terminal before creating the PR

roli-wg commented on todoapp/src/components/DeletePopUP.vue:
```diff
@@ -0,0 +1,168 @@
+  left: 0;
+  top: 0;
+  bottom: 0;
+  right: 0;
```
We have a shorthand for setting the top, left, bottom and right to 0 using [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset) you can achieve the same in one line `inset: 0`

EmanuelRusuu committed:
Removed unnecessary code/ improved styling

EmanuelRusuu committed:
Added possibility to edit the title / description of a todo
